lib/google/gax/errors.rb

```diff
@@ -67,6 +56,22 @@ def metadata
         return nil unless cause&.respond_to? :metadata
         cause.metadata
       end
+
+      ##
+      # @private Wrap GRPC error with associated GAX error object.
```

Comment:
I'm not sure we should exclude this from the public API. It might be useful for some kinds of customization and I don't think it adds much complexity to expose and document it. Any reason you want to keep it?

Reply:
I can't think of a use case for why a user would want to wrap a `GRPC::BadStatus` error with a `Google::Gax::GaxError`. It seems like internal logic to me.

Reply:
Interceptors was the primary use case that I had in mind, but agreed it will be rare.

IIRC Ads was also creating `GaxError` instances in order to extract the status details so we should probably keep that helper exposed.

Reply:
K, reverted.